### PR TITLE
feat: add support for pylint on python stack (#71)

### PR DESCRIPTION
**Pull request by marianesoares**
Add support for Pylint to the lint stage on python stack (#71) 

Tested in a python project

### CODE REVIEW

**Review by oesgalha — CHANGES_REQUESTED** [review]
Can you remove the changes at `.github/workflows/**`?
They should enter in another PR, because they aren't related to #71.

---

The `pylint` linter is very opinionated and shouldn't be used/suggested as a default linter. Per #71 it should only be included in the CI pipeline if the project uses it.

**Review by joao10lima — APPROVED** [review]

File: core/templates/github/python/lint.yaml [hunk 1]
```diff
@@ -20,11 +20,17 @@ jobs:
...
+<% if (!it.linters.pylint.isDependency && it.linters.pylint.hasConfig) { -%>
+      - run: python -m pip install pip pylint
+<% } -%>
 
 <% if (it.linters.flake8) { -%>
       # Adapts Flake8 to run with the Black formatter, using the '--ignore' flag to skip incompatibilities errors
       # Reference: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html?highlight=other%20tools#id1
       - run: <%= it.packageManager.commands.run _%> flake8 --ignore E203,E501,W503 .
 <% } -%>
+<% if (it.linters.pylint && (it.linters.pylint.hasConfig || it.linters.pylint.isDependency)) { -%>
```
Comment (oesgalha) [inline core/templates/github/python/lint.yaml:32]: Can you simplify this conditional?

Try to make it look like the preceding flake8 conditional: 
```
if (it.linters.pylint)
```

**Review by oesgalha — COMMENTED** [review]
Well done! I've tested here and it works. 👏 

I've written down some suggestions to improve it, can you take a look?

File: core/plugins/stack/python/linters.ts [hunk 2]
```diff
@@ -24,6 +28,27 @@ export const introspect: IntrospectFn<Linters> = async (context) => {
...
   }
+  if (!pyLintInfo) {
+    linters.pylint = {
+      isDependency: false,
+      hasConfig: false,
+      name: "pylint",
+    };
+  } else if (pyLintInfo.isDependency) {
+    logger.debug("detected Pylint");
+    console.log(pyLintInfo);
```
[the remark this answers is not in the document]
Comment (marianesoares) [reply core/plugins/stack/python/linters.ts]: Done

File: core/templates/github/python/lint.yaml (hunk 1 above)
Comment (marianesoares) [reply core/templates/github/python/lint.yaml]: Done!

**Review by oesgalha — APPROVED** [review]
Nice job! 😁

**Pull request merged**